api/pkg/apis/v1alpha1/managers/activations/activations-manager.go

```diff
@@ -323,15 +331,35 @@ func (t *ActivationsManager) ReportStageStatus(ctx context.Context, name string,
 	return nil
 }
 
-func mergeStageStatus(activationState *model.ActivationState, current model.StageStatus) error {
+func mergeStageStatus(ctx context.Context, activationState *model.ActivationState, current model.StageStatus) error {
 	if current.Outputs["__site"] == nil {
 		// The StageStatus is triggered locally
 		if activationState.Status.StageHistory == nil {
 			activationState.Status.StageHistory = make([]model.StageStatus, 0)
 		}
+
+		currentBytes, _ := json.Marshal(current)
+		if len(currentBytes) > StageOutputMaxSize {
```

Comment:
The assumption of checking output here is other parts are pretty small. However, is it possible input map is also very large? For example, one of the input is $output.

Reply:
Added truncate also for inputs.
Other fields of current are all string or bool, so we would have to deal with them.